### ISSUE TITLE
Need conda-forge package for 3.12

### ISSUE THREAD

**Issue by michaelaye** [Anaconda-Platform/nb_conda_kernels#230]
At the moment there doesn't seem to be a package for Python 3.12?

```bash
❯ mamba install nb_conda_kernels astropy kalasiris

Looking for: ['nb_conda_kernels', 'astropy', 'kalasiris']

conda-forge/linux-64                                        Using cache
conda-forge/noarch                                          Using cache

Pinned packages:
  - python 3.12.*


warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
Could not solve for environment specs
The following packages are incompatible
└─ nb_conda_kernels is installable with the potential options
   ├─ nb_conda_kernels [1.0.3|2.0.0|...|2.2.3] would require
   │  └─ python [2.7* |>=2.7,<2.8.0a0 ], which can be installed;
   ├─ nb_conda_kernels [1.0.3|2.0.0] would require
   │  └─ python 3.4* , which can be installed;
   ├─ nb_conda_kernels [1.0.3|2.0.0|2.1.0|2.1.1] would require
   │  └─ python 3.5* , which can be installed;
   ├─ nb_conda_kernels [2.0.0|2.1.0|2.1.1] would require
   │  └─ python 3.6* , which can be installed;
   ├─ nb_conda_kernels [2.1.1|2.2.0|...|2.3.1] would require
   │  └─ python >=3.6,<3.7.0a0 , which can be installed;
   ├─ nb_conda_kernels [2.1.1|2.2.0|...|2.3.1] would require
   │  └─ python >=3.7,<3.8.0a0 , which can be installed;
   ├─ nb_conda_kernels [2.2.2|2.2.3|2.2.4|2.3.0|2.3.1] would require
   │  └─ python >=3.8,<3.9.0a0 , which can be installed;
   ├─ nb_conda_kernels [2.3.0|2.3.1] would require
   │  └─ python >=3.9,<3.10.0a0 , which can be installed;
   ├─ nb_conda_kernels 2.3.1 would require
   │  └─ python >=3.10,<3.11.0a0 , which can be installed;
   └─ nb_conda_kernels 2.3.1 would require
      └─ python >=3.11,<3.12.0a0 , which can be installed.
```

**Comment by fcollonval** [comment]
I think you can transfer the issue on the following repository: https://github.com/conda-forge/nb_conda_kernels-feedstock

**Comment by michaelaye** [comment]
I guess, even so, it looks like issues aren't looked at over there.

**Comment by philip-khor** [comment]
If it helps, I recall from glancing at the logs (now not accessible unfortunately) that the error for 3.12 builds had to do with versioneer

**Comment by philip-khor** [comment]
Running `pip install -e .` on py312 returns:

```py
      C:\Users\Admin\Documents\GitHub\nb_conda_kernels\versioneer.py:421: SyntaxWarning: invalid escape sequence '\s'
        LONG_VERSION_PY['git'] = '''
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\Admin\Documents\GitHub\nb_conda_kernels\setup.py", line 6, in <module>
          version=versioneer.get_version(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Admin\Documents\GitHub\nb_conda_kernels\versioneer.py", line 1480, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "C:\Users\Admin\Documents\GitHub\nb_conda_kernels\versioneer.py", line 1412, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Admin\Documents\GitHub\nb_conda_kernels\versioneer.py", line 342, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?

```



Perhaps useful: python-versioneer/python-versioneer#375

**Comment by cipri-tom** [comment]
@fcollonval We asked on the feedstock repository, and they are ready to upgrade the compatibility list. But cannot do it as of now, since it doesn't work on 312. The maintainer is wondering if the package is still maintained ? https://github.com/conda-forge/nb_conda_kernels-feedstock/pull/49#issuecomment-1810674292 

Would this versioneer error be easy to fix ? Thanks !

**Comment by effigies** [comment]
The versioneer issue is resolved, but an old versioneer is vendored here. Upgrade with `pipx run versioneer install`.

See https://github.com/python-versioneer/python-versioneer/blob/master/UPGRADING.md for any migration issues.

**Comment by basnijholt** [comment]
I opened https://github.com/Anaconda-Platform/nb_conda_kernels/pull/236 after realizing that a naive update in the conda-forge feedstock (https://github.com/conda-forge/nb_conda_kernels-feedstock/pull/53) didn't work.

Would be glad to have anyone look at my PR (#236).

**Comment by mcg1969** [comment]
fixed by #237 (and thanks to @basnijholt for #236, which drove it).
